executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -78,11 +87,18 @@ pub fn split_out_machines<'a, T: FieldElement>(
             base_identities.iter().cloned().partition(|i| {
                 // The identity's left side has at least one machine witness, but
                 // all referenced witnesses are machine witnesses.
-                // Note that expressions of "simple" polynomial identities are stored
-                // in i.left.selector, so those are covered as well.
                 // For lookups, any lookup calling from the current machine belongs
                 // to the machine; lookups to the machine do not.
-                let all_refs = &refs_in_selected_expressions(&i.left) & (&all_witnesses);
+                let all_refs = match i {
+                    Identity::Polynomial(identity) => {
```

Comment:
Maybe extract that into a function?

Reply:
done, with all_children

Reply:
That was wrong so done, just extracting as suggested